lib/commands.coffee

```diff
@@ -51,12 +52,26 @@ class Commands
       atom.notifications.addError error.message, {detail: error.stack, dismissable: true}
 
   nextError: ->
-    if not @messages or (next = @messages.next()).done
-      next = (@messages = @linter.views.getMessages().values()).next()
-    return if next.done # There's no errors
-    message = next.value
-    return unless message.filePath
-    return unless message.range
+    if @index?
+      @index++
+    else
+      @index = 0
+    messages = @linter.views.messages
+    message = messages[@index %% messages.length]
```

Comment:
Yes, it is also taking the dividend independent modulo so that negative indexes work as well, taken from the coffeescript site:
```coffee
-7 %% 5 == 3 # n %% 5 is always between 0 and 4
```

I could add a comment, I could also add a function, but it seems weird to me to increase the `@index` in this method but then to call another method to use the index.

How about an `getCurrentMessage(index)` method?